### PR TITLE
Add "contributing to Nixpkgs" tutorial

### PR DESCRIPTION
**Pull request by proofconstruction**
While writing *Packaging Existing Software with Nix*, I originally had a section at the end where the reader is walked through preparing the package for submission to Nixpkgs. This is a cleaned-up version of that which maintains continuity with the other tutorial, and should help get people contributing faster.

There is a lot more to be said on this topic, but probably all of *that* belongs in a recipe instead.

### CODE REVIEW

File: source/tutorials/contributing-to-nixpkgs.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+$ nix-build -A icat
+this derivation will be built:
+  /nix/store/kf7na78dhwhfrgxjaj27gjkqs4b3n7wz-icat.drv
+...
+/nix/store/vr2vk8z8839l5j6gra0qlyxrh5sarmh4-icat
+```
+
+The build completed as expected, and produced a `result` in the current directory.
+
+The package is now ready for submission to Nixpkgs!
```
Comment (zmitchell) [inline source/tutorials/contributing-to-nixpkgs.md:182]: You should try running the package first :)

**Review by zmitchell — CHANGES_REQUESTED** [review]
Generally I think you can cut down how much of this is rehashing the previous tutorial, but other than that I'm a fan of the content.

I think this belongs in the learning journey section given that it's a direct follow up to the learning journey tutorial it refers to.

File: source/tutorials/contributing-to-nixpkgs.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+---
+
+
+(contributing-to-nixpkgs)=
+
+# Contributing to Nixpkgs
+
+[Nixpkgs](https://github.com/nixos/nixpkgs) is one of the largest public repositories by contributions and *the* largest single collection of software packages available on the web.
+
+In this tutorial, which follows {ref}`the tutorial on packaging existing software <packaging-existing-software>`, you will learn how to prepare newly-packaged software for submission to Nixpkgs, becoming a Nixpkgs contributor in the process.
```
[the remark this answers is not in the document]
Comment (yukiisbored) [reply source/tutorials/contributing-to-nixpkgs.md:15]: Unlike links, this uses the `packaging-existing-software` reference and will work no matter where it is placed in the file system as long as the reference exists somewhere.

File: source/tutorials/contributing-to-nixpkgs.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+## Adding to Nixpkgs
+You can now clone a local copy of `nixpkgs`, and move your derivation into an appropriate location within that source tree.
+
+```console
+$ git clone https://github.com/nixos/nixpkgs
+```
+
+`icat` is a console tool which also displays graphics, so there are a few places within Nixpkgs that you could reasonably move it.
+
+There are currently two other similar tools already in `nixpkgs`, both of which are stored in `pkgs/applications/graphics`, so this is a good candidate for a home for `icat` too.
```
Comment (infinisil) [inline source/tutorials/contributing-to-nixpkgs.md:139]: You should use [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme) for new packages now! No need to edit `all-packages.nix` either that way :)

File: source/tutorials/contributing-to-nixpkgs.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+The package is now ready for submission to Nixpkgs!
+
+## Next Steps
+With the package building successfully, you're almost ready to contribute your work to Nixpkgs.
+
+Be sure to read the [contributing documentation](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) to understand the current submission conventions before proceeding.
+
+:::{note}
+Before contributing your first package, you *must* add your information to [`nixpkgs/maintainers/maintainers-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix), following the instructions [here](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).
```
Comment (infinisil) [inline source/tutorials/contributing-to-nixpkgs.md:190]: This is a better docs link: https://github.com/NixOS/nixpkgs/tree/master/maintainers#readme

**Review by infinisil — CHANGES_REQUESTED** [review]

**Review by bzm3r — COMMENTED** [review]
I like it! I learned just from reading the PR, so yeah, it would be really nice to have this on nix.dev!

I think the changes suggested by @infinisil and @zmitchell are quite useful. So, once those get merged....

File: source/tutorials/contributing-to-nixpkgs.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+## Tags, Revisions, and Releases
+An earlier version of the `icat.nix` definition created in the Packaging Existing Software tutorial used the `master` revision of the [upstream `icat` repository](https://github.com/atextor/icat), which would be suitable for individual use but isn't good practice for submission to Nixpkgs.
+
+It is much better to use a fixed revision, corresponding to a particular release version of the software, so maintainers (perhaps you!) can easily check whether the package should be updated.
+
+This was already done in the previous tutorial, but as a reminder, here are the steps:
+1. Check the upstream repository for tags or releases. On GitHub, these are available at `https://github.com/<owner>/<repo>/tags` and `https://github.com/<owner>/<repo>/releases` respectively.
+2. Choose the latest tag or release, and write it into a string literal as the value of the `rev` attribute passed to the [fetcher](https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers).
+3. Use `nix-prefetch-url` or a similar tool to fetch, unpack, and hash the files marked by the tag or release you selected.
```
[the remark this answers is not in the document]
Comment (bzm3r) [reply source/tutorials/contributing-to-nixpkgs.md:25]: Yes! I strongly agree with this.

**Comment by infinisil** [comment]
@bzm3r Since the original author seems busy, could you create a new PR taking the feedback into account?

**Comment by proofconstruction** [comment]
Decided in [this comment](https://github.com/NixOS/nix.dev/pull/831#issuecomment-1879545180) to close this and merge anything useful into the `CONTRIBUTING.md` in nixpkgs, etc.

**Pull request closed without merging**